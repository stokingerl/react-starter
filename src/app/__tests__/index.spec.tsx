import React from 'react'
import { fireEvent, render, screen } from '@testing-library/react'
import { App } from '../index'


describe('src / app', () => {
  it('should match snapshot', () => {
    const app = render(<App />)
    expect(app.baseElement).toMatchSnapshot()
  })

  it('should render counter and button', () => {
    render(<App />)

    expect(screen.getByRole('heading')).toHaveTextContent('0')
    expect(screen.getByRole('button')).toHaveTextContent('press me')
  })

  it('should increment counter', () => {
    render(<App />)

    fireEvent.click(screen.getByRole('button'))

    expect(screen.getByRole('heading')).toHaveTextContent('1')
  })
})
