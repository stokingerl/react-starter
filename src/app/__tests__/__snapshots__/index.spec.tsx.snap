// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src / app should match snapshot 1`] = `
<body>
  <div>
    <h1>
      0
    </h1>
    <button
      class="counterButton"
    >
      press me
    </button>
  </div>
</body>
`;
